In [26]:
# Importing required libraries
import pandas as pd
import matplotlib.pyplot as plt

# Importing clean csv file for analysis
savings_df_clean = pd.read_csv('savings_data_clean.csv', parse_dates=['Month'])
savings_df_clean.head()

,Month,Category,Type,Amount,Notes
0,2024-01-01,Salary,Income,45000.0,Monthly salary
1,2024-01-01,Freelance,Income,8000.0,Tutoring income
2,2024-01-01,Rent,Expense,12000.0,House rent
3,2024-01-01,Food,Expense,6500.0,Groceries and meals
4,2024-01-01,Transport,Expense,2200.0,Bus and fuel


In [27]:
# Checking for missing values/ duplicates
savings_df_clean.isnull().sum()

Month       0
Category    0
Type        0
Amount      0
Notes       0
dtype: int64

In [28]:
savings_df_clean.duplicated().sum()

np.int64(0)

##### Calculating monthly totals and savings rate

In [29]:
# Checking the type of money

savings_df_clean['Type'].unique()

<StringArray>
['Income', 'Expense', 'Savings']
Length: 3, dtype: str

In [41]:
# Calculating monthly totals
# Seperating income, expense and savings.
income_df = savings_df_clean[savings_df_clean['Type'] == 'Income']
expense_df = savings_df_clean[savings_df_clean['Type'] == 'Expense']
savings_df = savings_df_clean[savings_df_clean['Type'] == 'Savings']

#Grouping them by month and sum
monthly_income = income_df.groupby('Month')['Amount'].sum().rename('Total_income')
monthly_expense = expense_df.groupby('Month')['Amount'].sum().rename('Total_expense')
monthly_savings = savings_df.groupby('Month')['Amount'].sum().rename('Total_savings')

# Mergint into one summary table
monthly = pd.concat([monthly_income, monthly_expense, monthly_savings], axis=1)
print(monthly)

            Total_income  Total_expense  Total_savings
Month                                                 
2024-01-01       53000.0        25200.0        27800.0
2024-02-01       50000.0        26200.0        23800.0
2024-03-01       57000.0        27000.0        29000.0
2024-04-01       51500.0        27800.0        23700.0
2024-05-01       57000.0        25100.0        31900.0
2024-06-01       55500.0        25900.0        35700.0
2024-07-01      106000.0        30500.0        27500.0
2024-08-01       52500.0        25000.0        28500.0
2024-09-01       59000.0        27600.0        31400.0
2024-10-01       56000.0        30000.0        32600.0
2024-11-01       54000.0        26400.0        27700.0
2024-12-01       63000.0        32700.0        29300.0


In [45]:
# Calculating Savings rate

# Savings rate = (Savings / Income) * 100
monthly['Savings_Rate(%)'] = (monthly['Total_savings'] / monthly['Total_income'] * 100).round(2)

print("\nMonthly Savings Rate:")
print(monthly[['Total_income', 'Total_savings', 'Savings_Rate(%)']])


Monthly Savings Rate:
            Total_income  Total_savings  Savings_Rate(%)
Month                                                   
2024-01-01       53000.0        27800.0            52.45
2024-02-01       50000.0        23800.0            47.60
2024-03-01       57000.0        29000.0            50.88
2024-04-01       51500.0        23700.0            46.02
2024-05-01       57000.0        31900.0            55.96
2024-06-01       55500.0        35700.0            64.32
2024-07-01      106000.0        27500.0            25.94
2024-08-01       52500.0        28500.0            54.29
2024-09-01       59000.0        31400.0            53.22
2024-10-01       56000.0        32600.0            58.21
2024-11-01       54000.0        27700.0            51.30
2024-12-01       63000.0        29300.0            46.51


In [51]:
# Highest saving rate
monthly['Savings_Rate(%)'].max()

np.float64(64.32)

##### The month of `July`has the highest savings rate of `64.32`

#####  Building category breakdown and running balance